core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -435,12 +459,35 @@ object ConsumerGroupCommand extends Logging {
       )
     }
 
-    protected def getLogEndOffset(topicPartition: TopicPartition): LogEndOffsetResult = {
+    protected def getLogEndOffset(topicPartition: TopicPartition): LogOffsetResult = {
       val consumer = getConsumer()
       consumer.assign(List(topicPartition).asJava)
       consumer.seekToEnd(List(topicPartition).asJava)
       val logEndOffset = consumer.position(topicPartition)
-      LogEndOffsetResult.LogEndOffset(logEndOffset)
+      LogOffsetResult.LogOffset(logEndOffset)
+    }
+
+    protected def getLogStartOffset(topicPartition: TopicPartition): LogOffsetResult = {
+      val consumer = getConsumer()
+      consumer.assign(List(topicPartition).asJava)
+      consumer.seekToBeginning(List(topicPartition).asJava)
+      val logStartOffset = consumer.position(topicPartition)
+      LogOffsetResult.LogOffset(logStartOffset)
+    }
+
+    protected def getLogTimestampOffset(topicPartition: TopicPartition, timestamp: Long): LogOffsetResult = {
+      val consumer = getConsumer()
+      consumer.assign(List(topicPartition).asJava)
+      val offsetForTimestamp = Option(consumer.offsetsForTimes(Map(topicPartition -> timestamp.asInstanceOf[java.lang.Long]).asJava))
```

Comment:
Also, why are we wrapping the result of `offsetsForTimes` in an `Option`? It can never be `null`.